src/scheduler/resv_info.c

```diff
@@ -1636,92 +1656,148 @@ confirm_reservation(status *policy, int pbs_sd, resource_resv *unconf_resv, serv
 	return rconf;
 }
 
+/**
+ * @brief determine if a nspec superchunk/chunk has unavailable nodes 
+ * 		and checks for running jobs on the chunk
+ * @param[in] resv - reservation to check
+ * @param[in] chunk_ind - index of the chunk start
+ * 
+ * @return int
+ * @retval 1 - running jobs, no unavailable nodes
+ * @retval 0 no unavailable nodes, no running jobs
+ * @retval -1 unavailable nodes, but no running jobs on the chunk
+ * @retval -2 unavilable nodes, running jobs within the chunk
+ * 
+ */
+int check_down_running(resource_resv *resv, int chunk_ind)
+{
+	int i, j, k;
+	int ret = 0;
```

Comment:
Can you please add a check to make sure resv is a reservation and not a job. Maybe also change the name of the function to say that it is only meant for reservation.